harvester/harvester
# Move support bundle image to chart

**Problem:**
We set the default support bundle image value in the code. It's hard to maintain the version between harvester and harvester-installer.

**Solution:**
Move support bundle image value to harvester chart, so we can get the default support bundle image version from the chart. If users would like to override it, they can still set the value in the `support-bundle-image` setting.

**Related Issue:**
https://github.com/harvester/harvester/issues/1727

**Test plan:**

1. Check the controller can get support bundle image version from App.

* Set following yaml in `spec.chart.values` of `harvester-system/harvester` app:
```
support-bundle-kit:
  image:
    imagePullPolicy: Always
    repository: rancher/support-bundle-kit
    tag: v0.0.4
```
* Create a support bundle.
* Check image pull policy, repository, and tag in support bundle pod are the same.
* Check default value of `support-bundle-image` setting is `{"repository":"rancher/support-bundle-kit","tag":"v0.0.4","imagePullPolicy":"Always"}`.

2. Check the controller can merge support bundle image value in App and ManagedChart.

* Following above case, set following yaml in `spec.values` of `fleet-local/harvester` managed chart:
```
support-bundle-kit:
  image:
    imagePullPolicy: IfNotPresent
```
* Create a support bundle.
* Check image pull policy in support bundle pod is `IfNotPresent`.
* Check default value of `support-bundle-image` setting is `{"repository":"rancher/support-bundle-kit","tag":"v0.0.4","imagePullPolicy":"IfNotPresent"}`.

3. Check value from users can override the default value in `support-bundle-image` setting

* Set value in `support-bundle-image` setting as `{"repository":"rancher/support-bundle-kit","tag":"v0.0.5","imagePullPolicy":"IfNotPresent"}`.
* Create a support bundle.
* Check image tag is `v0.0.5` in support bundle pod.
